src/main/java/ontos/Ontos.java

```diff
@@ -0,0 +1,88 @@
+package ontos;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import ontos.commands.Command;
+import ontos.parser.Parser;
+import ontos.storage.SaveManager;
+import ontos.task.TaskList;
+import ontos.ui.Ui;
+
+/**
+ * Represents the main application that manages the task list.
+ * It handles user interactions, command parsing, task management, and saving/loading tasks.
+ */
+public class Ontos {
+    /** The user interface used to print outputs from the chatbot in the CLI */
+    private Ui ui;
+    /** The list of tasks the chatbot manages */
+    private TaskList tasks;
+    /** The Savemanager object used to read and write to the save file */
+    private SaveManager saveManager;
+
+    /**
+     * Constructs an Ontos instance with the specified save location.
+     *
+     * @param saveLocation The location of the save file relative to the project's root directory.
+     */
+    public Ontos(String saveLocation) {
+        Path projectRoot = Paths.get("").toAbsolutePath().getParent().getParent().getParent();
+        this.ui = new Ui();
```

Comment:
The coding standard recommends avoiding unnecessary use of `this` with fields.

https://se-education.org/guides/tutorials/intellijCodeStyle.html